docs/2.guide/3.going-further/10.runtime-config.md

```diff
@@ -49,21 +49,19 @@ Instead of passing non-serializable objects or functions into your application f
 
 The most common way to provide configuration is by using [Environment Variables](https://medium.com/chingu/an-introduction-to-environment-variables-and-how-to-use-them-f602f66d15fa).
 
-::alert{type=info}
-Nuxt CLI has built-in [dotenv](https://github.com/motdotla/dotenv) support in development mode and when running `nuxi build` and `nuxi generate`.
+::alert{type=warning}
+Nuxi CLI has built-in support for reading your `.env` file in development, build and generate. But when you run your built server, **your `.env` file will not be read**.
 
-In addition to any process environment variables, if you have a `.env` file in your project root directory, it will be automatically loaded **at build, dev, and generate time**, and any environment variables set there will be accessible within your `nuxt.config` file and modules.
+:ReadMore{link="/docs/guide/directory-structure/env"}
+::
 
-When updating `.env` in development mode, the Nuxt instance is automatically restarted to apply new values to the `process.env`.
+Runtime config values are automatically replaced by matching environment variables at runtime. There are two key requirements.
 
-However, **after your server is built**, you are responsible for setting environment variables when you run the server. Your `.env` file will not be read at this point. How you do this is different for every environment. On a Linux server, you could pass the environment variables as arguments using the terminal `DATABASE_HOST=mydatabaseconnectionstring node .output/server/index.mjs`. Or you could source your env file using `source .env && node .output/server/index.mjs`.
+1. Your desired variables must be defined in your `nuxt.config`. This ensures that arbitrary environment variables are not exposed to your application code.
 
-Note that for a purely static site, it is not possible to set runtime configuration config after your project is prerendered.
-::
+1. Only a specially-named environment variable can override a runtime config property. That is, an uppercase environment variable starting with `NUXT_` which uses `_` to separate keys and case changes.
```

Comment:
💡 Idea: Can we create an interactive component that converts env name to NUXT_ variable? (i personally even often need a moment to think how to map!)

Reply:
I wonder if it's possible to do via type hinting 🤔

Reply:
I have a working prototype. PR incoming.

Reply:
~> https://github.com/nuxt/nuxt/pull/18652